Kernel/VM/MemoryManager.cpp

```diff
@@ -433,14 +444,25 @@ RetainPtr<Region> MemoryManager::allocate_kernel_region(size_t size, String&& na
 RetainPtr<PhysicalPage> MemoryManager::allocate_physical_page(ShouldZeroFill should_zero_fill)
 {
     InterruptDisabler disabler;
-    if (1 > m_free_physical_pages.size()) {
-        kprintf("FUCK! No physical pages available.\n");
+
+    if (m_free_physical_pages.is_empty()) {
+        for (auto region : m_physical_regions) {
```

Comment:
We can iterate using auto& here rather than auto. The difference is subtle but basically in this case it means we don’t create a copy of the Retained<PhysicalRegion> in each iteration. A copy is really just inc/dec of the retain count but it’s good to make yourself aware of auto vs auto&.

Basically, auto can sometimes make copies, auto* and auto& will never make copies.